eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -6,8 +6,28 @@ parameters:
   # invocations
   DependenciesVariableName: vcpkg.deps
 
+  # Ref at which to check out (can be commit SHA or tag in the form of
+  # `tags/<tag>`)
+  VcpkgVersion: c6d69984edceacbe880fad20ddc8305a7ec6fb8c
+
 steps:
-  # Mac OS specific requirements
+  # Set VCPKG_INSTALLATION_ROOT location for cache on Mac OS hosts
+  - task: Bash@3
+    inputs:
+      targetType: inline
+      workingDirectory: $(Agent.TempDirectory)
+      script: |
+        echo "##vso[task.prependpath]$(Agent.TempDirectory)/vcpkg"
+        echo "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT;]$(Agent.TempDirectory)/vcpkg"
+    displayName: MacOS - Set VCPKG_INSTALLATION_ROOT
+    condition: >-
+      and(
+        succeeded(),
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
```

Comment:
I am not familiar with yml, can you explain to me what this is checking for and why it is necessary?

Essentially, we are checking that the DependenciesVariableName isn't an empty string, correct. When would it be empty?
```
parameters.DependenciesVariableName != ''
```

Reply:
We are checking that the variable whose name is described in `parameters.DependenciesVariableName` (`vcpkg.deps` by default) is not empty. Another way to say it might be: 

`variables[$variableName] != ''`